src/query/pipeline/transforms/src/processors/transforms/transform_sort_merge.rs

```diff
@@ -51,23 +46,13 @@ impl Compactor for SortMergeCompactor {
         "SortMergeTransform"
     }
 
-    fn interrupt(&self) {
-        self.aborting.store(true, Ordering::Release);
-    }
-
     fn compact_final(&self, blocks: &[DataBlock]) -> Result<Vec<DataBlock>> {
         if blocks.is_empty() {
             Ok(vec![])
         } else {
-            let aborting = self.aborting.clone();
-            let aborting: Aborting = Arc::new(Box::new(move || aborting.load(Ordering::Relaxed)));
+            let block = DataBlock::concat(blocks)?;
```

Comment:
We should only concat the sorted columns.

1. concat the sorted columns to a block
2. apply sort_indices to this block and get the indices
3. use `take_indices` to take from `blocks` and get the result